osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/valueobjects/CombinedDeviceModelCode.java

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: Copyright Contributors to the GXF project
+//
+// SPDX-License-Identifier: Apache-2.0
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.valueobjects;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Container for combined DeviceModelCodes that contains a gateway device model code and multiple
+ * channel based device model codes in the order of the channels 1-4. The format is :
+ * GatewayDeviceModelCode,Channel1DeviceModelCode,Channel2DeviceModelCode,Channel3DeviceModelCode,Channel4DeviceModelCode
+ */
+public class CombinedDeviceModelCode {
+
+  public static final String SEPERATOR = ",";
+
+  private final String gatewayDeviceModelCode;
+  private final Map<Integer, String> channelBasedDeviceModelCodes;
+
+  private CombinedDeviceModelCode() {
+    this.gatewayDeviceModelCode = "";
+    this.channelBasedDeviceModelCodes = new HashMap<>();
+  }
+
+  private CombinedDeviceModelCode(
+      final String gatewayDeviceModelCode,
+      final Map<Integer, String> channelBasedDeviceModelCodes) {
+    this.gatewayDeviceModelCode = gatewayDeviceModelCode;
+    this.channelBasedDeviceModelCodes = channelBasedDeviceModelCodes;
+  }
+
+  public static CombinedDeviceModelCode parse(final String combinedDeviceModelCodes) {
+    if (combinedDeviceModelCodes != null) {
+      final String[] codes = combinedDeviceModelCodes.split(SEPERATOR);
```

Comment:
Should we check if `codes` always has 5 elements?

Reply:
Yes we should check for correctness of this combinedDeviceModelCodes. So I added a check for this, named hasCombinedDeviceModelCodes.